### PR TITLE
Task/argis refactor stdr

### PR DESCRIPTION
**Pull request by AlvaroVega**
Comes from  https://github.com/telefonicaid/fiware-cygnus/pull/1860 

This PR will be used to align original branch to current master

New Arcgis sink to avoid licence problems with SDK file.

[cygnus-ngsi][feature][NGSIArcgisSink] create NGSIArcgisSink #1672
[cygnus-ngsi][NameMapping] Allow deep Json attribute mapping. #1856

### CODE REVIEW

**Comment by fgalan** [comment]
Note that at a8230ef the PR has the same diff checksum than the original one, i.e. +6,783 −1,313

**Comment by AlvaroVega** [comment]
Changes introduced by [cygnus-ngsi][NameMapping] Allow deep Json attribute mapping. #1856
are a different feature and should be discussed in another PR, so will be removed from this PR 

cygnus-common/src/main/java/com/telefonica/iot/cygnus/containers/NameMappings.java 
cygnus-ngi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSINameMappingsInterceptor.java
cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/containers/NameMappingsTest.java 
cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/interceptors/NGSINameMappingsInterceptorTest.java
doc/cygnus-ngsi/installation_and_administration_guide/name_mappings.md

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[cygnus-ngsi][ArcgisFeatureTableSink]New sink to persist Arcgis data. (#1672)
+[cygnus-arcgis][new backend]New Rest based backend to persist Arcgis data. (#1672)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:2]:
```suggestion
[cygnus-ngsi][ArcGis] New sink (rework) to persist Arcgis data (#1672)
[cygnus-common][ArcGis] New Rest based backend to persist Arcgis data. (#1672)
```

Not sure, but I think is closer to the "changelog style", as seen in https://github.com/telefonicaid/fiware-cygnus/releases?after=FIWARE_7.8.2

Not sure about cygnus-commons, but I'd say that backends are included in tha part, isn't it?

File: .gitignore [hunk 2]
```diff
@@ -14,4 +14,5 @@ cygnus-twitter/bin
 .classpath
 .project
 .factorypath
-*/.settings
\ No newline at end of file
+*/.settings
+.*
```
Comment (fgalan) [inline .gitignore:18]:
```suggestion
*/.settings
.*
```

as I understand that files like `.gitignore` itself should be included in the repo :)

**Review by AlvaroVega — COMMENTED** [review]
LGTM

File: cygnus-common/pom.xml [hunk 3]
```diff
@@ -16,6 +16,12 @@
     </properties>
 
     <dependencies>
```
Comment (fgalan) [inline cygnus-common/pom.xml:18]: Weird indent in the added XML fragments...

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_arcgis_featuretable_sink.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# <a name="top"></a>NGSIArcGisSink
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_arcgis_featuretable_sink.md:1]: This file has been renamed. Review references to it in other parts of the documenation (with grep it should be easy), so links can be fixed.

At least I think that .mkdocs.yml should be adjusted.

File: CHANGES_NEXT_RELEASE [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[cygnus-ngsi][ArcGis] New sink (rework) to persist Arcgis data (#1672)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]: Mabye even better:

```suggestion
[cygnus-ngsi][NGSIArcgisFeatureTableSink] New sink to persist Arcgis data, removing the old ArgGis sink (#1672)
```

File: cygnus-common/pom.xml (hunk 3 above)
Comment (AlvaroVega) [reply cygnus-common/pom.xml:18]: fixed

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSISink.java [hunk 6]
```diff
@@ -30,25 +44,8 @@
...
-import java.util.Date;
-import org.apache.flume.Channel;
-import org.apache.flume.Context;
-import org.apache.flume.Event;
-import org.apache.flume.EventDeliveryException;
-import org.apache.flume.Sink.Status;
-import org.apache.flume.Transaction;
-import org.apache.flume.ChannelException;
-import org.apache.flume.conf.Configurable;
-import org.apache.log4j.MDC;
```
Comment (IvanHdzC) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSISink.java:51]: I don't quite understand why this lines were removed. I thing the code will not compile without this lines. 

They should be aggregated.

Comment (IvanHdzC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSISink.java:51]: My mistake. I didn't look closely, the lines were moved from its place.

NTC

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_arcgis_featuretable_sink.md (hunk 4 above)
Comment (AlvaroVega) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_arcgis_featuretable_sink.md:1]: done

File: cygnus-common/pom.xml [hunk 7]
```diff
@@ -141,6 +141,15 @@
...
             <plugins>
+                <!-- SKIP TEST -->
+	        <plugin>
+	          <groupId>org.apache.maven.plugins</groupId>
+	          <artifactId>maven-surefire-plugin</artifactId>
+	          <version>2.12.4</version>
+	          <configuration>
+		    <skipTests>true</skipTests>
+	          </configuration>
+	        </plugin>
```
Comment (fgalan) [inline cygnus-common/pom.xml:152]: Really needed? Which new code is using it?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/arcgis/NGSIArcgisFeatureTable.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    private long timeoutSecs = 60;
+    private Date lastPersist = new Date();
+
+    /**
+     * 
+     * @param featureServiceUrl
+     * @param username
+     * @param password
+     * @param getTokenUrl
+     * @param b
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/arcgis/NGSIArcgisFeatureTable.java:45]:
```suggestion
     * @param timeoutSecs
```

Weird... maybe there are another cases of "not fully replaced JavaDoc templates" like this?

File: cygnus-common/pom.xml (hunk 7 above)
Comment (AlvaroVega) [reply cygnus-common/pom.xml:152]: Used by com.telefonica.iot.cygnus.backends.arcgis.CredentialRestApiTest

As you can see in https://travis-ci.org/github/telefonicaid/fiware-cygnus/jobs/744408999 without this plugin we obtain:
```
Results :

Failed tests:   getObjectIdTest(com.telefonica.iot.cygnus.backends.arcgis.FeatureTest): Encuentra id cuando no debería haberlo.

  checkResponseTest(com.telefonica.iot.cygnus.backends.arcgis.RestApiTest): Response should be ok.

Tests in error: 

  testUserCredential(com.telefonica.iot.cygnus.backends.arcgis.CredentialRestApiTest): Can´t parse token generation url, MalformedURLException.

  getSetObjectId(com.telefonica.iot.cygnus.backends.arcgis.FeatureTest): java.lang.Integer cannot be cast to java.lang.Long

Tests run: 220, Failures: 2, Errors: 2, Skipped: 4

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project cygnus-common: There are test failures.

[ERROR] 

[ERROR] Please refer to /home/travis/build/telefonicaid/fiware-cygnus/cygnus-common/target/surefire-reports for the individual test results.

[ERROR] -> [Help 1]

[ERROR] 

[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.

[ERROR] Re-run Maven using the -X switch to enable full debug logging.

[ERROR] 

[ERROR] For more information about the errors and possible solutions, please read the following articles:

[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

<<<<<<<<<<<< Run cygnus-ngsi tests >>>>>>>>>>>>
```

Comment (fgalan) [reply cygnus-common/pom.xml]: Ok. NTC

The intent stills looking a bit ugly. I have tried to fixed, without success. It seems to be some kind of tabs vs space issue, but it doesn't worth the time to keep trying.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/arcgis/NGSIArcgisFeatureTable.java (hunk 8 above)
Comment (fgalan) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/arcgis/NGSIArcgisFeatureTable.java]: Fixed in 6cc4a5a

File: .gitignore (hunk 2 above)
Comment (fgalan) [reply .gitignore]: Fixed in 704e06e

**Review by IvanHdzC — APPROVED** [review]
LGTM

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_arcgis_featuretable_sink.md [hunk 9]
```diff
@@ -103,39 +134,36 @@ A configuration example could be:
     ...
     cygnus-ngsi.sinks.arcgis-sink.type = com.telefonica.iot.cygnus.sinks.NGSIArcGisSink
```
Comment (IvanHdzC) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_arcgis_featuretable_sink.md:135]:
```suggestion
    cygnus-ngsi.sinks.arcgis-sink.type = com.telefonica.iot.cygnus.sinks.NGSIArcgisFeatureTableSink
```

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by AlvaroVega** [comment]
LGTM

**Comment by IvanHdzC** [comment]
LGTM

**Pull request merged**